src/cmake-tools.ts

```diff
@@ -51,12 +51,30 @@ const log = logging.createLogger('main');
 const BUILD_LOGGER = logging.createLogger('build');
 const CMAKE_LOGGER = logging.createLogger('cmake');
 
-enum ConfigureType {
+export enum ConfigureType {
   Normal,
   Clean,
   Cache,
 }
 
+export enum ConfigureTrigger {
+  api = "Triggered by calling configure commands via the CMake Tools API",
```

Comment:
I would rather not have long complicated strings for this.  Using the enum name as the string is best, I think.